### ISSUE TITLE
Barcode, QRCode not generating

### ISSUE THREAD

**Issue by alecsorenpedersen** [alexandercerutti/passkit-generator#151]
Mac V12.6

Node v20.2.0

## Description

I have created a pass, everything works except the QR code & barcode this is my pass.json

```
{
	"passTypeIdentifier": "pass.link.help-id",
	"serialNumber": "001",
	"teamIdentifier": "JL9PCED3DG",
	"description": "EMERGENCYS SORT IT",
	"organizationName": "help-id",
	"logoText": "HELP ID",
	"foregroundColor": "rgb(255, 255, 255)",
	"backgroundColor": "rgb(60, 65, 76)",
	"barcode": {
		"format": "PKBarcodeFormatQR",
		"message": "PLACEHOLDER_EMERGENCYCONTACTPHONE",
		"messageEncoding": "iso-8859-1"
	},
	"generic": {
		"primaryFields": [{ "key": "name", "value": "PLACEHOLDER_NAME" }],
		"secondaryFields": [
			{ "key": "medicalConditions", "value": "PLACEHOLDER_MEDICALCONDITIONS" },
			{ "key": "medications", "value": "PLACEHOLDER_MEDICATIONS" }
		],
		"auxiliaryFields": [
			{ "key": "medicalDevices", "value": "PLACEHOLDER_MEDICALDEVICES" },
			{ "key": "dateOfBirth", "value": "PLACEHOLDER_DATEOFBIRTH" }
		],
		"backFields": [
			{ "key": "allergies", "value": "PLACEHOLDER_ALLERGIES" },
			{
				"key": "emergencyContactName",
				"value": "PLACEHOLDER_EMERGENCYCONTACTNAME"
			},
			{
				"key": "emergencyContactRelationship",
				"value": "PLACEHOLDER_EMERGENCYCONTACTRELATIONSHIP"
			},
			{
				"key": "emergencyContactPhone",
				"value": "PLACEHOLDER_EMERGENCYCONTACTPHONE"
			}
		]
	}
}
```

Tried so much, got to the point where even ChatGPT4 told me to ask you as it didn't know

**Comment by alexandercerutti** [comment]
Hey @alecsorenpedersen, thanks for using passkit-generator!

Sorry, but your post doesn't specify enough details about how you composed your pass, so if you could share your code, I might be helpful!

You are saying that your barcode is not showing? Where are you trying to opening your pass?

Let me know!

**Comment by alexandercerutti** [comment]
I think to understand what is happening.
You know, `pass.json` property `barcode` is actually deprecated by Apple and, therefore, I do not support it anymore. When `pass.json` is loaded, `barcode` fails validation and gets excluded.

You should actually hardcode `barcodes` ([here is the link to Apple Wallet Documentation](https://developer.apple.com/documentation/walletpasses/pass), please note that it is an array) or use `pass.setBarcodes()` API to let it handle for you.

Let me know if this solves your issue!

**Comment by alexandercerutti** [comment]
Oh, some suggestions about your code:

1) You can use `pass.mimeType` instead of writing it on your own.
2) You can use `pass.getAsStream()` instead of converting it by yourself (I already do that 😄)

I hope you appreciate ;)

**Comment by alecsorenpedersen** [comment]
I got it working thank you so much

**Comment by alexandercerutti** [comment]
Hey there @alecsorenpedersen, I saw you attempted to delete part of this conversation.

It's okay for what concerns the code (even if it's almost the same everywhere). For what concerns the first message, revisions were still visible, so I proceeded into recovering one and delete the others. While recovering, I have hidden most of the information about your pass.json, which was useless to the issue thread. I preferred to recover at least the main content to give some context for whoever will read it in the future! I hope you don't mind!
